File: 3174-clear-digits/main.go
func clearDigits(s string) string {
	var ans []rune

	for _, char := range s {
		if unicode.IsDigit(char) && len(ans) > 0 {
			ans = ans[:len(ans)-1]
		} else {
			ans = append(ans, char)
		}
	}

	return string(ans)
}

File: 3174-clear-digits/main.cpp
class Solution {
public:
    string clearDigits(string s) {
        vector<char> ans;

        for (char ch : s) {
            if (isdigit(ch) && !ans.empty()) {
                ans.pop_back();
            } else {
                ans.push_back(ch);
            }
        }

        return string(ans.begin(), ans.end());
    }
};
